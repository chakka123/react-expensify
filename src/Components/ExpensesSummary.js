import React from 'react';
import { connect } from 'react-redux';

import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <p>Viewing {props.expenses.length} expense totalling {numeral(expensesTotal(props.expenses)).format('$0,u0.00')}.</p>
)

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});


export default connect(mapStateToProps)(ExpensesSummary);