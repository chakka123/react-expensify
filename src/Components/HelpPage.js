import React from 'react';

const HelpPage = () => (
    <div>
        This the help page
    </div>
);

export default HelpPage;