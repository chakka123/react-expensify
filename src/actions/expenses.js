import uuid from 'uuid';


export const addExpense = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => {
    return {
        type: 'ADD_EXPENSE',
        expense: {
            id: uuid(),
            description,
            note,
            amount,
            createdAt
        }
    }
};

export const removeExpense = ({ id = undefined } = {}) => {
    return {
        type: 'REMOVE_EXPENSE',
        id: id
    }
};

export const editExpense = (id = undefined, updates) => {
    return {
        type: 'EDIT_EXPENSE',
        id: id,
        updates
    }
};

