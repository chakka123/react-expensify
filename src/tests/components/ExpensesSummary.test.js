import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../Components/ExpensesSummary';
import expenses from '../fixtures/expenses';


test('Should render ExpensesSummary correctly', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />)
    expect(wrapper).toMatchSnapshot();
});