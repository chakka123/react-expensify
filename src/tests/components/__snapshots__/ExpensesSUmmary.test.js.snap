// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly 1`] = `
<p>
  Viewing 
  3
   expense totalling 
  $24,195.00
  .
</p>
`;
